misc/util.py

```diff
@@ -157,12 +157,30 @@ def ensure_test_db(connection_string, keep_temps=False):
 
 
 # whoever calls this will need to import boto and get the s3 client
-def firstFromS3Prefix(s3, bucket, prefix, desired_filename, outdir=None, outpath=None):
+def firstFromS3Prefix(
+    s3, bucket, prefix, desired_filename, outdir=None, outpath=None
+) -> bool:
+    haystack = []
+    found_needle = False
+
+    def report():
+        print(
+            f"""Hello from firstFromS3Prefix() !!!!
```

Comment:
this has the flavor I'd expect from a `--verbose` level debug print. Do we really want it on all the time?

Reply:
Didn't spot this comment in time before merging. I would advocate for it (not so much the silly Hello, but more the print out of the haystack being searched). I find it useful to see which S3 objects actually were being considered. I'm open to further discussion.